src/osd/ReplicatedPG.cc

```diff
@@ -10462,6 +10423,11 @@ void ReplicatedPG::hit_set_trim(RepGather *repop, unsigned max)
   for (unsigned num = updated_hit_set_hist.history.size(); num > max; --num) {
     list<pg_hit_set_info_t>::iterator p = updated_hit_set_hist.history.begin();
     assert(p != updated_hit_set_hist.history.end());
+    if (!p->using_gmt) {
+      // do not delete hit_set using the localtime in case any peer is in a
+      // different timezone
+      continue;
```

Comment:
Also, you can't just continue; here since the rest of this loop assumes that all previous entries have been trimmed.